### PR TITLE
Controller error handlers

### PR DESCRIPTION
**Pull request by ghost**
#261

### CODE REVIEW

File: controllers/bonfire.js [hunk 1]
```diff
@@ -48,14 +48,16 @@ exports.returnNextBonfire = function(req, res) {
...
-    req.user.save();
+    req.user.save(function(err) {
+        if (err) return next(err);
+    });
 
     var uncompletedBonfires = req.user.uncompletedBonfires;
 
     var displayedBonfires =  Bonfire.find({'_id': uncompletedBonfires[0]});
     displayedBonfires.exec(function(err, bonfire) {
         if (err) {
```
Comment (BerkeleyTrue) [inline controllers/bonfire.js]: Subsequent async methods must be done within the first async callback. So all the code after user.save must be within it's callback.

**Comment by jameskopacz** [comment]
Like that?

**Comment by BerkeleyTrue** [comment]
LGTM. Needs to be test. I don't think I will be able to until the weekend, though.

**Comment by BerkeleyTrue** [comment]
@jameskopacz Can you do me a favor? Do you know how to squash commits with `git rebase -i` ? I would like this to be one single commit. If you aren't comfortable do that we can pair on the weekend and I can walk you through it.

**Comment by jameskopacz** [comment]
Hmm, I rebased and squashed the changes in SourceTree. Does that look correct?

**Comment by BerkeleyTrue** [comment]
No, you would end up with a single commit in this PR.

**Comment by jameskopacz** [comment]
What about now, specifically: https://github.com/jameskopacz/freecodecamp/commit/1b1e1f5fc30c9f9f6fc5e4e8958f21a1c81d3053

**Comment by BerkeleyTrue** [comment]
You should pushed (it will need to be a force push) it to the same branch that this PR comes from. And change the commit message so it is more succinct "handle callback errors for controller/user and ...". The end result will be this PR will be updated to show just one commit instead of multiple commits with the same message.

**Comment by BerkeleyTrue** [comment]
To be more specific. 

* do: git rebase -i <the hash for the first commit in this pr>
* now squash all the commits and rewrite the message [here](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) is a good post on proper commit messages.
* Complete the rebase
* No your local and remote branch of jameskopacz-patch-261 should differ.
*do git push -f to remote branch. This will wipe out the history and replace it with the local one so please use with caution. 
* github will recognize the change and update this PR automatically. Now it will show one commit with a succinct and imperative message. but the end changes to the files will  be the same. 

This practice of limiting PR's to one or a few commits is common in open source community and helps keep project history clean and searchable.

**Comment by jameskopacz** [comment]
YEAH! I think it worked!

**Comment by BerkeleyTrue** [comment]
:+1: Isn't git awesome!

**Comment by jameskopacz** [comment]
I wish it would of been done in SourceTree but using the command line did feel pretty cool, especially when it worked!

**Comment by BerkeleyTrue** [comment]
yeah ditch SoureTree. Do everything in command line. Using stuff like that ends up handicapping you. Git is meant for the command line. It's ok to use it for visual stuff, though I still do that in vim/command line.

**Comment by jameskopacz** [comment]
What would be visual stuff?

**Comment by BerkeleyTrue** [comment]
diffs and visualizing all the different branches.

**Comment by jameskopacz** [comment]
Ohh yeah, cool. That makes sense.

**Comment by QuincyLarson** [comment]
Hey @BerkeleyTrue can you QA this real quick and verify it's good for production?

**Comment by BerkeleyTrue** [comment]
@jameskopacz @QuincyLarson merged

**Pull request merged**